modules/apps/portal/portal-upgrade-impl/src/main/java/com/liferay/portal/upgrade/internal/report/UpgradeReport.java

```diff
@@ -121,6 +130,112 @@ private String _getDialectInfo() {
 			StringPool.PERIOD, db.getMinorVersion(), StringPool.NEW_LINE);
 	}
 
+	private String _getDocLibSize() {
+		if (_dlStore.equals("com.liferay.portal.store.db.DBStore")) {
+			return StringPool.BLANK;
+		}
+		else if (_dlStore.contains("FileSystemStore")) {
+			if ((_rootDir == null) &&
+				_dlStore.equals(
+					"com.liferay.portal.store.file.system." +
+						"AdvancedFileSystemStore")) {
+
+				return StringBundler.concat(
+					"\"rootDir\" was not set, unable to determine the size of ",
+					"the document library", StringPool.NEW_LINE);
+			}
+
+			final AtomicLong length = new AtomicLong(0);
+
+			String docLibPath =
+				PropsValues.LIFERAY_HOME + "/data/document_library";
+
+			if (_rootDir != null) {
+				docLibPath = _rootDir + "/data/document_library";
+			}
+
+			try {
+				Files.walkFileTree(
```

Comment:
@SamZiemer, have you tried the performance of this? Do we have an estimation about how long it takes for a large number of files?

To simplify the code, do you think FileUtils.sizeOfDirectory() can be an option? We already use commons-io in portal-impl

Reply:
I just tested using this option, and it works just as well as my previous option. Regarding the performance, each was similar on my machine (around 1-2 seconds), however using FileUtils.sizeOfDirectory() is much cleaner. I will add that change in a new commit on top the current commit.

Reply:
perfect, thanks!